### PR TITLE
feat: Pass type != transaction condition where we don't want transactions

### PR DESCRIPTION
**Pull request by lynnagara**
This is an alternative to https://github.com/getsentry/sentry/pull/15118
which added a dataset argument (defaulting to Events) on an eventstore
query. Here we are forcing the correct dataset decision by adding a
condition "type != transaction" anywhere we do not want to look at
transactions (everything except Discover 2).

When we expand `get_events` and `get_events_by_id` to support
transactions, we will need to add the same to the callers of those functions.

### CODE REVIEW

**Review by fpacifici — APPROVED** [review]

File: src/sentry/api/endpoints/organization_event_details.py [hunk 1]
```diff
@@ -91,7 +91,7 @@ def oldest_event_id(self, snuba_args, event):
 
     def _get_filter(self, snuba_args):
         return eventstore.Filter(
-            conditions=snuba_args["conditions"],
+            conditions=snuba_args["conditions"] + [["type", "!=", "transaction"]],
```
Comment (markstory) [inline src/sentry/api/endpoints/organization_event_details.py]: Don't we need to get transaction events when viewing them?

Comment (lynnagara) [reply src/sentry/api/endpoints/organization_event_details.py]: Oh.... transaction events also use this endpoint? In that case this won't work

Comment (lynnagara) [reply src/sentry/api/endpoints/organization_event_details.py]: Removed the condition from this endpoint, thanks

Comment (markstory) [reply src/sentry/api/endpoints/organization_event_details.py]: Yes this endpoint is used to get the payload to render the details modal.

**Review by markstory — APPROVED** [review]

**Pull request merged**